src/main/java/org/pac4j/vertx/handler/impl/RequiresAuthenticationHandler.java

```diff
@@ -138,8 +145,35 @@ public void handle(RoutingContext routingContext) {
 
     }
 
-    protected void unauthorized(VertxWebContext webContext, List<Client> currentClients) {
-        httpActionAdapter.handle(HttpConstants.UNAUTHORIZED, webContext);
+    @Override
+    protected void authorise(final User user, final RoutingContext context) {
+        if (! (user instanceof Pac4jUser)) {
+            throw new TechnicalException("Wrong user type in authorise");
+        }
+
+        Handler<AsyncResult<Boolean>> authHandler = res -> {
+            if (res.succeeded()) {
+                if (res.result()) {
+                    context.next();
+                } else {
+                    forbidden(context);
+                }
```

Comment:
Looks good to me. I'm looking forward to see everything in action (in the demo).